File: practice_project/custom_code/quotation.py
import frappe

def validate_quotation(doc, method):
    # Get Customer Type from Customer master
    if doc.customer:
        customer_type = frappe.get_value("Customer", doc.customer, "customer_type")

        if customer_type == "Individual":
            if not doc.customer_name:
                frappe.throw("Customer Name is mandatory for Individual customers")
            if not doc.custom_mobile_number:
                frappe.throw("Mobile Number is mandatory for Individual customers")

File: practice_project/public/js/quotation.js
frappe.ui.form.on("Quotation", {
    refresh: function(frm) {
        if (frm.doc.docstatus === 1 || frm.doc.docstatus === 0)  {   // Only show button if Quotation is submitted or draft
            frappe.db.get_value("Customer", frm.doc.customer, "customer_type", (r) => {
                if (r && r.customer_type === "Individual") {
                    frm.add_custom_button(__("Create Sales Invoice"), function() {
                        frappe.call({
                            method: "frappe.model.mapper.make_mapped_doc",
                            args: {
                                method: "erpnext.selling.doctype.quotation.quotation.make_sales_invoice",
                                source_name: frm.doc.name
                            },
                            callback: function(r) {
                                if (r.message) {
                                    var doc = frappe.model.sync(r.message)[0];
                                    frappe.set_route("Form", doc.doctype, doc.name);
                                }
                            }
                        });
                    }, __("Create"));
                }
            });
        }
    }
});
